### ISSUE TITLE
Build issue: An accessor cannot be declared in an ambient context.

### ISSUE THREAD

**Issue by ishwara-bhat** [chrisguttandin/multi-buffer-data-view#42]
I am using extendable-media-record package and multi-buffer-data-view is one of the dependencies. I get the below error. Moving to higher typescript is not an option for me.

Please let me know if any solution to this.

ERROR in node_modules/multi-buffer-data-view/build/es2018/classes/multi-buffer-data-view.d.ts:8:9 - error TS1086: An accessor cannot be declared in an ambient context.

8     get buffers(): ArrayBuffer[];
          ~~~~~~~
node_modules/multi-buffer-data-view/build/es2018/classes/multi-buffer-data-view.d.ts:9:9 - error TS1086: An accessor cannot be declared in an ambient context.

9     get byteLength(): number;
          ~~~~~~~~~~
node_modules/multi-buffer-data-view/build/es2018/classes/multi-buffer-data-view.d.ts:10:9 - error TS1086: An accessor cannot be declared in an ambient context.

10     get byteOffset(): number;

My versions:
Angular CLI: 8.3.21
Node: 10.16.0
OS: win32 x64
Angular: 8.2.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.21
@angular-devkit/build-angular     0.803.21
@angular-devkit/build-optimizer   0.803.21
@angular-devkit/build-webpack     0.803.21
@angular-devkit/core              8.3.21
@angular-devkit/schematics        8.3.21
@angular/cdk                      8.0.0
@angular/cli                      8.3.21
@angular/flex-layout              8.0.0-beta.26
@angular/material                 8.0.0
@ngtools/webpack                  8.3.21
@schematics/angular               8.3.21
@schematics/update                0.803.21
rxjs                              6.5.3
typescript                        3.5.3
webpack                           4.39.2

My package.json
{
  "name": "annular",
  "version": "1.4.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@agm/core": "1.0.0-beta.5",
    "@angular/animations": "^8.2.14",
    "@angular/cdk": "^8.0.0",
    "@angular/cli": "^8.3.21",
    "@angular/common": "^8.2.14",
    "@angular/compiler": "8.2.14",
    "@angular/core": "^8.2.14",
    "@angular/flex-layout": "8.0.0-beta.26",
    "@angular/forms": "8.2.14",
    "@angular/material": "^8.0.0",
    "@angular/platform-browser": "8.2.14",
    "@angular/platform-browser-dynamic": "8.2.14",
    "@angular/router": "8.2.14",
    "@babel/runtime": "^7.7.7",
    "@fortawesome/angular-fontawesome": "^0.5.0",
    "@fortawesome/fontawesome-svg-core": "^1.2.22",
    "@fortawesome/free-solid-svg-icons": "^5.10.2",
    "@ngu/carousel": "1.5.5",
    "@nicky-lenaers/ngx-scroll-to": "^2.0.0",
    "@swimlane/ngx-charts": "11.1.0",
    "@swimlane/ngx-datatable": "15.0.1",
    "@types/dom-mediacapture-record": "^1.0.2",
    "@types/node": "^12.7.5",
    "angular-agora-rtc": "^0.1.0",
    "angular-calendar": "0.27.8",
    "angular-in-memory-web-api": "0.8.0",
    "asyncawait": "^3.0.0",
    "aws-sdk": "^2.528.0",
    "chart.js": "^2.8.0",
    "d3": "^5.11.0",
    "d3-ng2-service": "^2.2.0",
    "date-fns": "1.30.1",
    "extendable-media-recorder": "^5.0.4",
    "hammerjs": "2.0.8",
    "httpclient": "^0.1.0",
    "leaflet-map": "0.2.1",
    "media-encoder-host": "^7.0.7",
    "ng2-charts": "^2.3.0",
    "ng2-dragula": "2.1.1",
    "ngx-audio-player": "^8.0.1",
    "ngx-pagination": "3.2.1",
    "ngx-perfect-scrollbar": "7.2.1",
    "ngx-quill": "5.2.0",
    "ngx-timer": "^1.0.6",
    "quill": "^1.3.7",
    "recorder-audio-worklet": "^4.0.36",
    "rxjs": "6.5.3",
    "standardized-audio-context": "^23.0.6",
    "timelines-chart": "^2.8.2",
    "tslib": "^1.10.0",
    "zone.js": "0.9.1"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^0.803.21",
    "@angular/compiler-cli": "^8.2.14",
    "@angular/language-service": "8.2.14",
    "@types/node": "~8.9.4",
    "@types/jasmine": "3.3.13",
    "@types/jasminewd2": "2.0.6",
    "codelyzer": "5.1.0",
    "jasmine-core": "3.4.0",
    "jasmine-spec-reporter": "4.2.1",
    "karma": "4.1.0",
    "karma-chrome-launcher": "2.2.0",
    "karma-coverage-istanbul-reporter": "2.0.5",
    "karma-jasmine": "2.0.1",
    "karma-jasmine-html-reporter": "1.4.2",
    "protractor": "5.4.2",
    "ts-node": "8.2.0",
    "tslint": "5.17.0",
    "typescript": "3.5.3"
  }
}

**Comment by chrisguttandin** [comment]
Hi @ishwara-bhat,

thanks for reporting this. I'm using this library in an Angular application as well. Unfortunately I updated it some days ago to v9.

The app is called [analog4all-provider](https://github.com/chrisguttandin/analog4all-provider).

If you want to keep using Angular v8, I guess you would have to at least update TypeScript to v3.6.x to get it to work. Once you do that you will get an error/warning from the CLI which you can disable as described [here](https://angular.io/guide/angular-compiler-options#disabletypescriptversioncheck).

I also had to manually add the minimal type definitions for the MediaRecorder. This is the file I came up with: https://github.com/chrisguttandin/analog4all-provider/blob/master/src/media-recorder.d.ts

Please let me know if that helps.

**Comment by ishwara-bhat** [comment]
Hi Christoph,
Thanks a lot. It greatly helped. It took me until ng serve. But I ran into type error during 'ng build --prod'
My updated config:-
My tfconfig.app.json 
..
"angularCompilerOptions": {
    "enableIvy": false,
    "disableTypeScriptVersionCheck" : true
  }

**This results in ng build --prod error** 
ERROR in TypeError: Cannot read property 'kind' of undefined
    at isParameterPropertyDeclaration (WorkAjs\node_modules\typescript\lib\typescript.js:13412:88)
    at Object.filter (WorkAjs\node_modules\typescript\lib\typescript.js:469:31)
    at transformClassMembers (WorkAjs\node_modules\typescript\lib\typescript.js:70699:20)
...

Please let me know if any suggestions. At the moment, I'm staying with v8 due to dependencies. Thanks in advance,
Ishwar

**Comment by chrisguttandin** [comment]
Oh sorry, I just realized that I responded to the wrong issue. But I think chrisguttandin/extendable-media-recorder#468 is the same issue right?

I just created a brand new Angular v8 project and just added `extendable-media-recorder` as dependency. I updated TypeScript to v3.6.x and set the `disableTypeScriptVersionCheck` option. I also added the `media-recorder.d.ts` file.

When I do all that running `ng build --prod` it works without throwing any error. Is it possible that some other dependency is causing that error in your project?

**Comment by ishwara-bhat** [comment]
Thanks a lot for the response. Yes, it was my mistake. I just repeated what you did. It works on fresh project. Sorry about raising this issue. Issue appears to be my configuration somewhere.

**Comment by chrisguttandin** [comment]
I'm glad it helped and I hope you can find the problem. Can we then close chrisguttandin/extendable-media-recorder#468 as well?

**Comment by ishwara-bhat** [comment]
Yes. In fresh project this is not a problem. I did not try this package explicitly though. Meanwhile, if possible kindly let me know answers to the following:
There is a package caleld @types/dom-mediacapture-record. I had installed it. It has Media type definitions. Would it clash with extendable-media-recorder? If it is not tested that way, it is fine. If any opinion, please let me know.
Thanks in advance.
Ishwar

**Comment by chrisguttandin** [comment]
Hi Ishwar, I never tested extendable-media-recorder with @types/dom-mediacapture-record but the types look fairly similar. I think it should work.

**Comment by ishwara-bhat** [comment]
Hi Christoph,

I solved my issue. Just thought I would let you know.:

1. As you specified, multi-buffer-data-view uses higher typescript and
disable version checking.
2. This higher version could run into memory issues during ng serve / ng
build.

Solution for #2 is to specify higher size :-

*node --max_old_space_size=8192 node_modules/@angular/cli/bin/ng build
--prod*

If someone else runs into issue while using  multi-buffer-data-view  with
higher typescript version, this could be a solution.

I have built is successfully and will test my functionality .

Regards
Ishwar
Regards
Ishwar
